### PR TITLE
Add snackbar component, and add it to the app

### PR DESCRIPTION
**Pull request by mewhhaha**
This pull request will add a snackbar component and reducer that can be called to show a snackbar with some kind of message (If you don't know what a snackbar is: https://www.polonel.com/snackbar/)
We could use this component to show relevant messages to the user, such as error messages

I've created two `SnackBar` components, one in `components/common` that is generic and one in `components` that uses the generic one but adds some extra functionality, like a dismiss button that dispatches a hide event.

If we so wish we could style the snackbar further in the future based  on importance level like https://codesandbox.io/s/5kq4jv6vpn but right now the styling is basic

*Motivation*
I tried opening a non-readable file in VS Code and they use a similar notification system

*How to use*
dispatch the action `showSnackBar` or `hideSnackBar` to show / hide the snackbar

If you want to try it out you can import the showSnackBar action to App.js and dispatch in componentDidMount like,

```javascript
import { showSnackBar } from './actions/dispatchActions';

...

class App extends Component {
    componentDidMount() {
        showSnackBar(this.props.dispatch, "IM A MESSAGE", "info", 10000);
    }

    ...
}
```

`showSnackBar` takes three extra arguments, message, level and fadeAfter
*message* is the message displayed
*level* is the type of snackbar message (this is unused right now, but could be useful in the future!)
*fadeAfter* is after how long the snackbar should fade. If not set or 0 then it won't disappear unless you press dismiss (or dispatch hideSnackBar)

`hideSnackBar` takes one extra argument, instant
*instant* if set to a truthy value will set fadeAfter to 0, which will result in the fadeout animation not running when hiding the snackbar

### CODE REVIEW

File: src/js/view/actions/dispatchActions.js [hunk 1]
```diff
@@ -59,3 +59,19 @@ export const handleCloseFile = dispatch => {
...
+
+export const hideSnackBar = (dispatch, instant) => {
+  dispatch({
+    type: 'snackbar_hide',
+    instant: instant
+  });
+};
+
+export const showSnackBar = (dispatch, message, level, fadeAfter) => {
+  dispatch({
```
Comment (AkarKhatab) [inline src/js/view/actions/dispatchActions.js]: borde vi ha 'message', 'level' och 'fadeAfter i ett 'data'-objekt så att vi skickar med endast 'type' och 'data' i dispatch som vi gör på andra ställen i koden?

File: src/js/view/components/SnackBar.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import { TextButton } from 'js/view/styledComponents/common/ButtonStyledComponents';
+import { hideSnackBar } from 'js/view/actions/dispatchActions';
+import {
+  Container,
+  Layout
+} from 'js/view/styledComponents/common/FlexBoxStyledComponents';
+import { connect } from 'react-redux';
+import CommonSnackBar from 'js/view/components/common/snackbar/SnackBar';
+
+function SnackBar(props) {
```
Comment (AkarKhatab) [inline src/js/view/components/SnackBar.js]: arrow-body-function istället?

File: src/js/view/components/common/snackbar/SnackBar.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React, { useState, useEffect } from 'react';
+import { SnackBarContainer } from 'js/view/styledComponents/common/SnackBarStyledComponents';
+
+export default function SnackBar(props) {
```
Comment (AkarKhatab) [inline src/js/view/components/common/snackbar/SnackBar.js]: arrow-body-function istället?

File: src/js/view/components/SnackBar.js (hunk 2 above)
Comment (mewhhaha) [reply src/js/view/components/SnackBar.js]: Kollade upp och det stod att det inte skulle vara någon skillnad i detta fallet så fixar!

File: src/js/view/components/common/snackbar/SnackBar.js (hunk 3 above)
Comment (mewhhaha) [reply src/js/view/components/common/snackbar/SnackBar.js]: Sure!

File: src/js/view/actions/dispatchActions.js (hunk 1 above)
Comment (mewhhaha) [reply src/js/view/actions/dispatchActions.js]: Jag ändrar så att det bundlas i ett dataobjekt, för denna pull requesten, så kan vi bestämma i framtiden hur vi vill ha det. Jag vet inte om vi behöver bundla det som ett dataobjekt.

**Pull request merged**